File: uponor/drivers/thermostat/device.js
'use strict';

const Homey = require('homey');
const { getVars, rawToCelsius, toInt } = require('../../lib/uponor');

const DEFAULT_POLL_INTERVAL_S = 60;

module.exports = class ThermostatDevice extends Homey.Device {
  async onInit() {
    const { controller, thermostat } = this.getStore();
    this._prefix = `C${controller}_T${thermostat}`;

    this._onSettingsChanged = (key) => {
      if (key === 'poll_interval') this._startPolling();
    };
    this.homey.settings.on('set', this._onSettingsChanged);

    await this._poll();
    this._startPolling();
  }

  async onUninit() {
    if (this._interval) this.homey.clearInterval(this._interval);
    this.homey.settings.off('set', this._onSettingsChanged);
  }

  _startPolling() {
    if (this._interval) this.homey.clearInterval(this._interval);
    const seconds = this.homey.settings.get('poll_interval') || DEFAULT_POLL_INTERVAL_S;
    this._interval = this.homey.setInterval(() => this._poll(), seconds * 1000);
  }

  async _poll() {
    const { host } = this.getStore();
    try {
      const vars = await getVars(host);
      const temp = rawToCelsius(vars[`${this._prefix}_room_temperature`]);
      if (temp !== null) {
        await this.setCapabilityValue('measure_temperature', temp).catch(this.error);
      }

      if (this.hasCapability('measure_humidity')) {
        const rh = toInt(vars[`${this._prefix}_rh`]);
        if (rh !== null) {
          await this.setCapabilityValue('measure_humidity', rh).catch(this.error);
        }
      }

      await this.setAvailable();
    } catch (err) {
      this.error('Poll failed:', err.message);
      await this.setUnavailable(err.message).catch(() => {});
    }
  }
};

File: uponor-smatrix_x-265+r-208/drivers/thermostat/device.py
from homey.device import Device

from ...lib.uponor import get_vars, raw_to_celsius, to_int

DEFAULT_POLL_INTERVAL_S = 60


class ThermostatDevice(Device):
    async def on_init(self):
        await super().on_init()
        store = self.get_store()
        self._prefix = f"C{store['controller']}_T{store['thermostat']}"
        self._interval = None

        self.homey.settings.on("set", self._on_setting_set)
        self._start_polling()  # the SDK's interval fires once immediately

    async def on_uninit(self):
        self.homey.clear_interval(self._interval)
        self.homey.settings.remove_listener("set", self._on_setting_set)

    def _on_setting_set(self, key):
        if key == "poll_interval":
            self._start_polling()

    def _start_polling(self):
        self.homey.clear_interval(self._interval)
        seconds = self.homey.settings.get("poll_interval") or DEFAULT_POLL_INTERVAL_S
        self._interval = self.homey.set_interval(self._poll, int(seconds) * 1000)

    async def _poll(self):
        try:
            v = await get_vars(self.get_store()["host"])
            temp = raw_to_celsius(v.get(f"{self._prefix}_room_temperature"))
            if temp is not None:
                await self.set_capability_value("measure_temperature", temp)

            if self.has_capability("measure_humidity"):
                rh = to_int(v.get(f"{self._prefix}_rh"))
                if rh is not None:
                    await self.set_capability_value("measure_humidity", rh)

            await self.set_available()
        except Exception as err:
            self.error("Poll failed:", err)
            await self.set_unavailable(str(err))


homey_export = ThermostatDevice
